Sources/AST/Type.swift

```diff
@@ -122,6 +127,12 @@ public struct Type: ASTNode {
     }
   }
 
+  var isSelfType: Bool {
```

Comment:
Why don't we use a 
`public private(set) var isSelfType`?
Then we don't have the duplication with `representsSelfType` but we have the same semantics.

Reply:
Actually, scratch that. Why don't you match the rawType to the self? Once we replace it the fact that we had a self will be irrelevant? Similar to `isCurrencyType`.

Reply:
I didn't know the first is valid syntax, but looks much neater. Will implement the second option.